TrackletAlgorithm/AllStubMemory.h

```diff
@@ -186,6 +186,27 @@ class AllStub : public AllStubBase<ASType>
     data_.range(kASBendMSB,kASBendLSB) = bend;
   }
 
+  std::string getBitStr() {
+    std::string str = decodeToBits(getR(),AllStubBase<ASType>::kASRSize);
+    str += "|"+decodeToBits(getZ(),AllStubBase<ASType>::kASZSize);
+    str += "|"+decodeToBits(getPhi(),AllStubBase<ASType>::kASPhiSize);
+    // Different get method for DISKPS?  Where is it?     
+    //     str += "|"+decodeToBits(getAlpha(),AllStubBase<ASType>::kASAlphaSize);
+    str += "|"+decodeToBits(getBend(),AllStubBase<ASType>::kASBendSize);
+    return str;
+  }
+
+  // TO DO: This belongs in some sort of helper class.
+  std::string decodeToBits(unsigned int field, unsigned int size) const {
+    unsigned int valtmp = field;
+    std::string str = "";
+    for(unsigned int i=0; i< size; i++) {
+      str = ((valtmp & 1) ? "1" : "0") + str;
+      valtmp >>= 1;
+    }
+    return str;
+  }
```

Comment:
@carriganm95 Would it make sense to make this a static method of `MemoryTemplate`? That would avoid having this code duplicated multiple times.

Reply:
does this need to be in a class at all? as far as I can tell it can just be a function.

Reply:
No it doesn't need to be in a class, but I thought `MemoryTemplate` would be a good place for it, since it's only used within the memory classes.

Reply:
I'm not commenting on what file it should sit in, just suggesting that it can be a simple function rather than a member function of some class.

Reply:
At least in my mind, it seems intimately tied to the `MemoryTemplate` class, which is why I proposed making it a static method of that class.

Reply:
I did make some changes to put this function in MemoryTemplate. Brian also pointed out that because this function is used when we write memories to file it could go in FileReadUtility.h as a simple function

Reply:
One thing I hadn't considered when I said that I would put this function (decodeToBits) in FileReadUtility.h is that the getBitStr() functions in AllStubMemory.h, etc. rely on this function and those files cannot directly include FileReadUtility.h. I could just use 

#include ../TestBenches/FileReadUtility.h

but I am thinking this is not good practice. Another suggestion from Brent was to just create a new file in TrackletAlgorithm called "GlobalFunctions.h" and include it there. Let me know if you have a preference on how this is handled and I can change it.